x-pack/filebeat/input/gcppubsub/otel_test.go

```diff
@@ -0,0 +1,171 @@
+	fbNameSpace := fmt.Sprintf("%x", uuid.Must(uuid.NewV4()))
+
+	type options struct {
+		Namespace    string
+		ESURL        string
+		Username     string
+		Password     string
+		Subscription string
+	}
+
+	gcpConfig := `filebeat.inputs:
+- type: gcp-pubsub
+  project_id: test-project-id
+  topic: test-topic-foo
+  subscription.name:  {{ .Subscription }}
+  credentials_file: "testdata/fake.json"
+
+output:
+  elasticsearch:
+    hosts:
+      - {{ .ESURL }}
+    username: {{ .Username }}
+    password: {{ .Password }}
+    index: logs-integration-{{ .Namespace }}
+
+queue.mem.flush.timeout: 0s
+setup.template.enabled: false
+processors:
+    - add_host_metadata: ~
+    - add_cloud_metadata: ~
+    - add_docker_metadata: ~
+    - add_kubernetes_metadata: ~
+`
+
+	// start filebeat in otel mode
+	filebeatOTel := integration.NewBeat(
+		t,
+		"filebeat-otel",
+		"../../filebeat.test",
+		"otel",
+	)
+
+	optionsValue := options{
+		ESURL:    fmt.Sprintf("%s://%s", host.Scheme, host.Host),
+		Username: user,
+		Password: password,
+	}
+
+	var configBuffer bytes.Buffer
+	optionsValue.Namespace = otelNamespace
+	optionsValue.Subscription = "test-subscription-otel"
+	require.NoError(t, template.Must(template.New("config").Parse(gcpConfig)).Execute(&configBuffer, optionsValue))
+
+	filebeatOTel.WriteConfigFile(configBuffer.String())
+
+	filebeatOTel.Start()
+	defer filebeatOTel.Stop()
+
+	// reset buffer
+	configBuffer.Reset()
+
+	optionsValue.Namespace = fbNameSpace
+	optionsValue.Subscription = "test-subscription-fb"
+	require.NoError(t, template.Must(template.New("config").Parse(gcpConfig)).Execute(&configBuffer, optionsValue))
+
+	// start filebeat
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+
+	filebeat.WriteConfigFile(configBuffer.String())
+	filebeat.Start()
+	defer filebeat.Stop()
+
+	// prepare to query ES
+	es := integration.GetESClient(t, "http")
+
+	rawQuery := map[string]any{
+		"query": map[string]any{
+			"match_phrase": map[string]any{
+				"input.type": "gcp-pubsub",
+			},
+		},
+		"sort": []map[string]any{
+			{"@timestamp": map[string]any{"order": "asc"}},
+		},
+	}
+
+	var filebeatDocs estools.Documents
+	var otelDocs estools.Documents
+	var err error
+
+	// wait for logs to be published
+	require.Eventually(t,
+		func() bool {
+			findCtx, findCancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer findCancel()
+
+			otelDocs, err = estools.PerformQueryForRawQuery(findCtx, rawQuery, ".ds-logs-integration-"+otelNamespace+"*", es)
+			assert.NoError(t, err)
+
+			filebeatDocs, err = estools.PerformQueryForRawQuery(findCtx, rawQuery, ".ds-logs-integration-"+fbNameSpace+"*", es)
+			assert.NoError(t, err)
+
+			return otelDocs.Hits.Total.Value >= 1 && filebeatDocs.Hits.Total.Value >= 1
+		},
+		3*time.Minute, 1*time.Second, fmt.Sprintf("Number of hits %d not equal to number of events %d", otelDocs.Hits.Total.Value, 1))
```

Comment:
[Blocker]
`Number of hits %d not equal to number of events %d` isn't what the test is asserting, it's checking both queries returned at least 1 document, but it could be more